### DISCUSSION TITLE
What is "install code-server" in Chrome browser?

### DISCUSSION THREAD

**Discussion by drmrbrewer** [coder/code-server#6755] [General]
I am running code-server in a Docker container and accessing it remotely in my Chrome browser.

In the window showing my code-server workspace, I see the following button to the right of the address bar at the top:

![image](https://github.com/coder/code-server/assets/9009792/adb0547f-58a6-42b2-b45a-946c4cf908da)

What exactly is this?

**Comment by lg-nineoneninethreenine** [comment]
Oh, that's the PWA install prompt. PWAs are progressive web apps. If you install that, then code-server runs as a separate window. It still runs on Chrome, but it appears to have its own standalone app.

Another PWA example is most ChromeOS apps that are pre-installed. They open as shortcuts on the Browser, but some apps like GitHub can open in separate windows while running on Chrome, that is, when they are actually "installed"(creating a shortcut and opening a separate window for that tab)

Basically this makes it look as though you installed code-server as an app, like VS Code. But still, it runs on Chrome. You _can_ try it, but if you don't like that, you can use it on the browser as you would.

**Reply by code-asher** [reply]
Yup, and to add to this, in our case nothing is actually installed/downloaded beyond the default browser caching; so it operates the same as usual, just in a separate window without a URL bar.

One good reason to use the PWA instead of a regular browser window is that keyboard shortcuts will work properly.  https://github.com/coder/code-server/blob/main/docs/FAQ.md#how-do-i-make-my-keyboard-shortcuts-work

And, it can be useful on mobile/tablet devices since it lets you put an icon on the home screen.

**Reply by drmrbrewer** [reply]
Well I never.  It does work really nicely as a PWA.  Thanks @im-coder-lg and @code-asher :-)

**Comment by barhun** [comment]
it'd be great if [chrome.sockets.tcpServer](https://developer.chrome.com/docs/apps/reference/sockets/tcpServer) was used to bind local ports for forwarding them to the instance that runs code-server. this way, the user would be able to use - for example - http://localhost:3000/ instead of https://code-server.example.com/proxy/3000/ to live view the app they're developing.

**Reply by barhun** [reply]
why did i write about this here? 'cause `tcpServer` can't be made use of on a webpage, it has to be used inside a web app installed via chrome.

[this](https://chromewebstore.google.com/detail/web-server-for-chrome/ofhbbkphhbklhfoeikjpcbhemlocgigb) app was the inspiration for me that made me think, maybe, the same functionality could help the contributors of code-server make this possible.

**Comment by Dipshandc** [comment]
Is there a way to open some files by default everytime i run code server in new instance ?
